sparksteps/__main__.py

```diff
@@ -244,6 +245,9 @@ def main():
         pricing_client = boto3.client('pricing', region_name=args_dict['aws_region'])
         args_dict = determine_prices(args_dict, ec2_client, pricing_client)
         cluster_config = cluster.emr_config(**args_dict)
+        with open(args_dict['configuration']) as json_file:  
```

Comment:
This will raise an `OSError` if `args_dict['configuration']` is not supplied - since `--configuration` is not a mandatory argument, we need to be able to handle this case.